### PR TITLE
Java bytecode build fixes

### PR DESCRIPTION
**Pull request by tautschnig**
This PR fixes three problems spotted while 1) compiling with SUB_IS_LIST set while having reverted #1742 (as done in #1860) and 2) building on AWS Codebuild (a75c4f0ac6835ce and friends got into the repository without PR).

### CODE REVIEW

**Review by smowton — APPROVED** [review]

File: COMPILING.md [hunk 1]
```diff
@@ -21,11 +21,11 @@ We assume that you have a Debian/Ubuntu or Red Hat-like distribution.
    The GNU Make needs to be version 3.81 or higher.
    On Debian-like distributions, do
    ```
-   apt-get install g++ gcc flex bison make git libwww-perl patch
+   apt-get install g++ gcc flex bison make git libwww-perl patch openjdk-7-jdk
```
Comment (peterschrammel) [inline COMPILING.md:24]: (Just a note for me to revert these changes as part of https://github.com/diffblue/cbmc/issues/1869).

**Review by peterschrammel — APPROVED** [review]
